server/src/main/java/org/apache/druid/discovery/DruidLeaderClient.java

```diff
@@ -213,6 +203,9 @@ public <T, H extends FullResponseHolder<T>> H go(Request request, HttpResponseHa
         ));
 
         request = withUrl(request, redirectUrl);
+      } else if (HttpResponseStatus.SERVICE_UNAVAILABLE.equals(responseStatus)
```

Comment:
I missed this but if there is a retry happening in response to 503/504, there will be no way to see that happening in logs. We should be logging a warning here. what do you think?

Reply:
Makes sense. Added a log.